File: 206_ReverseLinkedList.java
import java.util.*;
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode leftofhead = null;
        while (head != null){
            ListNode nextval = head.next; 
            head.next = leftofhead; 
            leftofhead = head; 
            head = nextval; 
        }
        return leftofhead;
    }
}

File: 206_ReverseLinkedList.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head){
        ListNode* revhead = nullptr;
        while (head != nullptr){
            ListNode* temp = head->next;
            head->next = revhead;
            revhead = head;
            head = temp;
        }
        return revhead;
    }
};